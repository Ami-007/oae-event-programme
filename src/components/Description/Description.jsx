import React, { useState } from "react";
import "./Description.scss";

const Description = (props) => {
  const { text, altText } = props;

  const [showAltText, setShowAltText] = useState(false);

  const getButtonActive = (isAltButton) => {
    if ((showAltText && isAltButton) || (!showAltText && !isAltButton)) {
      return "description__button--active";
    }
    return "";
  };

  return (
    <div className="description">
      {altText && (
        <div className="description__container">
          <button
            className={`description__button ${getButtonActive(false)}`}
            onClick={() => setShowAltText(false)}
          >
            Original
          </button>
          <button
            className={`description__button ${getButtonActive(true)}`}
            onClick={() => setShowAltText(true)}
          >
            Alternate
          </button>
        </div>
      )}
      <p>{showAltText ? altText : text}</p>
    </div>
  );
};

export default Description;
