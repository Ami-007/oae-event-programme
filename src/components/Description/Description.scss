@import "../../assets/sass/variables";

.description {
  &__container {
    display: flex;
    justify-content: center;
    gap: 1rem;
  }
  &__button {
    border: none;
    padding: 0.6rem 1rem;
    font-weight: 600;
    font-size: 0.9rem;
    color: $color-grey;
    &--active {
      background-color: var(--theme-color-primary);
      color: var(--theme-color-text);
      border: none;
      padding: 0.6rem 1rem;
      font-weight: 600;
    }
  }
}
