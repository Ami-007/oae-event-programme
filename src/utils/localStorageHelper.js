export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
