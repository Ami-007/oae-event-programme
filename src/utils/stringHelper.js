export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
