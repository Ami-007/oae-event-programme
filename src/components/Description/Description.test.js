import { screen, fireEvent } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import Description from "./Description";

test("Renders description component", () => {
  const { container } = customRender(
    <Description text="If this is your first Bach" />
  );
  expect(container).toMatchSnapshot();
});

test("If no alt text button shouldn't render", () => {
  customRender(<Description text="If this is your first Bach" />);
  const button = screen.queryByRole("button");
  expect(button).not.toBeInTheDocument();
});

test("If there is an alternate description buttons should render", () => {
  customRender(
    <Description
      text="If this is your first Bach"
      altText="alternate description"
    />
  );
  const button = screen.getByText("Alternate").closest("button");
  expect(button).toBeInTheDocument();
});

test("When alternate button is pressed alternate description renders", () => {
  customRender(
    <Description
      text="If this is your first Bach"
      altText="alternate description"
    />
  );
  const button = screen.getByText("Alternate");
  fireEvent.click(button);
  const altText = screen.getByText("alternate description");

  expect(altText).toBeInTheDocument();
});
