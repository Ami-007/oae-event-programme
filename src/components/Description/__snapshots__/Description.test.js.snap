// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders description component 1`] = `
<div>
  <div
    class="description"
  >
    <p>
      If this is your first Bach
    </p>
  </div>
</div>
`;
